src/wallet/api/schemas/walletApi.ts

```diff
@@ -0,0 +1,513 @@
+/**
+ * Copyright (c) Hathor Labs and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import { z } from 'zod';
+import { bigIntCoercibleSchema } from '../../../utils/bigint';
+
+/**
+ * Base response schema that all API responses extend from.
+ * Contains a success flag indicating if the operation was successful.
+ */
+const baseResponseSchema = z.object({
+  success: z.boolean(),
+});
+
+/**
+ * Schema for individual address information.
+ * Represents a single address in the wallet with its derivation index and transaction count.
+ */
+export const getAddressesObjectSchema = z.object({
+  address: z.string(), // Address in base58
+  index: z.number(), // derivation index of the address
+  transactions: z.number(), // quantity of transactions
+});
+
+/**
+ * Response schema for getting all addresses in the wallet.
+ */
+export const addressesResponseSchema = baseResponseSchema.extend({
+  addresses: z.array(getAddressesObjectSchema),
+});
+
+/**
+ * Response schema for checking if addresses belong to the wallet.
+ * Maps addresses to boolean values indicating ownership.
+ */
+export const checkAddressesMineResponseSchema = baseResponseSchema.extend({
+  addresses: z.record(z.boolean()), // WalletAddressMap
+});
+
+/**
+ * Schema for address information used in new address generation.
+ */
+export const addressInfoObjectSchema = z
+  .object({
+    address: z.string(),
+    index: z.number(),
+    addressPath: z.string(),
+    info: z.string().optional(),
+  })
+  .strict();
+
+/**
+ * Response schema for generating new addresses.
+ */
+export const newAddressesResponseSchema = baseResponseSchema.extend({
+  addresses: z.array(addressInfoObjectSchema),
+});
+
+/**
+ * Response schema for token details.
+ * Contains information about a token's name, symbol, total supply, and authorities.
+ */
+export const tokenDetailsResponseSchema = baseResponseSchema.extend({
+  details: z.object({
+    tokenInfo: z.object({
+      id: z.string(),
+      name: z.string(),
+      symbol: z.string(),
+    }),
+    totalSupply: bigIntCoercibleSchema,
+    totalTransactions: z.number(),
+    authorities: z.object({
+      mint: z.boolean(),
+      melt: z.boolean(),
+    }),
+  }),
+});
+
+/**
+ * Schema for token information.
+ */
+export const tokenInfoSchema = z.object({
+  id: z.string(),
```

Comment:
Check for 64 length and hex characters?
I included the native token case as well.

```suggestion
  id: z.string().regex(/^[a-fA-F0-9]{64}$|^00$/),
```

Reply:
Great, thanks!